### PR TITLE
Add Email types

### PR DESCRIPTION
**Pull request by nelsonjsduarte**
- EmailMessage
- EmailEvent
- EmailExportedHandler

### CODE REVIEW

**Comment by nelsonjsduarte** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by jasnell — APPROVED** [review]

File: types/defines/email.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
+ * A email message that is sent to a consumer Worker.
+ */
+interface EmailMessage<Body = unknown> {
+  /**
+   * A envelope from.
+   */
+  readonly from: string;
```
Comment (edevil) [inline types/defines/email.d.ts:8]: "Envelope From attribute of the email message."

File: types/defines/email.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * A envelope to.
```
Comment (edevil) [inline types/defines/email.d.ts:10]: "Envelope To attribute of the email message."

File: types/defines/email.d.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * A [Headers object](https://developer.mozilla.org/en-US/docs/Web/API/Headers).
+   */
+  readonly headers: Headers;
+  /**
+   * Stream of the email message content.
+   */
+  readonly raw: ReadableStream;
+  /**
+   * Size of the raw email message content.
```
Comment (edevil) [inline types/defines/email.d.ts:22]: "Size of the email message content."

File: types/defines/email.d.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  readonly rawSize: number;
+  /**
+   * Reject email message
```
Comment (edevil) [inline types/defines/email.d.ts:26]: "Reject this email message by returning a permanent SMTP error back to the connecting client including the given reason."

File: types/defines/email.d.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+   * @param reason The reject reason.
+   * @returns void
+   */
+  setReject(reason: String): void;
+  /**
+   * Forward email message to a verified destination address.
```
Comment (edevil) [inline types/defines/email.d.ts:32]: "Forward this email message to a verified destination address of the account."

File: types/defines/email.d.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+   * @param rcptTo Verified destination address.
+   * @param headers A [Headers object](https://developer.mozilla.org/en-US/docs/Web/API/Headers).
+   * @returns A promise that resolves when the email message is forward
```
Comment (edevil) [inline types/defines/email.d.ts:35]: "A promise that resolves when the email message is forwarded."

**Review by edevil — CHANGES_REQUESTED** [review]

**Review by edevil — APPROVED** [review]

**Pull request merged**